spring-integration-core/src/main/java/org/springframework/integration/router/RecipientListRouter.java

```diff
@@ -93,12 +93,15 @@ public void setChannels(List<MessageChannel> channels) {
 	 * @param recipients The recipients.
 	 */
 	public void setRecipients(List<Recipient> recipients) {
-		Assert.notEmpty(recipients, "recipients must not be empty");
+		Assert.notEmpty(recipients, "'recipients' must not be empty");
 		ConcurrentLinkedQueue<Recipient> originalRecipients = this.recipients;
 		this.recipients.clear();
 		this.recipients.addAll(recipients);
```

Comment:
This should be atomic - looks like it was broken by INT-2856; although not 

There is a small timing hole where there's no recipients, or an incomplete list.